pkg/scheduler/framework/runtime/framework.go

```diff
@@ -752,30 +752,29 @@ func (f *frameworkImpl) runFilterPlugin(ctx context.Context, pl framework.Filter
 }
 
 // RunPostFilterPlugins runs the set of configured PostFilter plugins until the first
-// Success or Error is met, otherwise continues to execute all plugins.
+// Success, Error or UnschedulableAndUnresolvable is met; otherwise continues to execute all plugins.
 func (f *frameworkImpl) RunPostFilterPlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod, filteredNodeStatusMap framework.NodeToStatusMap) (_ *framework.PostFilterResult, status *framework.Status) {
 	startTime := time.Now()
 	defer func() {
 		metrics.FrameworkExtensionPointDuration.WithLabelValues(postFilter, status.Code().String(), f.profileName).Observe(metrics.SinceInSeconds(startTime))
 	}()
 
-	statuses := make(framework.PluginToStatus)
 	// `result` records the last meaningful(non-noop) PostFilterResult.
 	var result *framework.PostFilterResult
 	for _, pl := range f.postFilterPlugins {
 		r, s := f.runPostFilterPlugin(ctx, pl, state, pod, filteredNodeStatusMap)
-		if s.IsSuccess() {
+		if s.IsSuccess() || s.IsUnresolvable() {
 			return r, s
 		} else if !s.IsUnschedulable() {
-			// Any status other than Success or Unschedulable is Error.
+			// Any status other than Success, Unschedulable or UnschedulableAndUnresolvable is Error.
 			return nil, framework.AsStatus(s.AsError())
 		} else if r != nil && r.Mode() != framework.ModeNoop {
 			result = r
 		}
-		statuses[pl.Name()] = s
+		status = status.Merge(s)
```

Comment:
Another approach is only appending reasons here, which seems reasonable if we always return unschedulable status without error.

Reply:
That is what I had in mind as well, I prefer not to expose functions unless it is needed... and status.Merge is not needed now, so I think we should remove all of that code.

Reply:
> Another approach is only appending reasons here, which seems reasonable if we always return unschedulable status without error.

+1. And this is the only place `status.Merge()` is called, so we can remove it as well as `statusPrecedence`.

Reply:
Updated.